### PR TITLE
Add method to compare configs.

### PR DESCRIPTION
**Pull request by fernandohds564**

### CODE REVIEW

**Review by anacso17 — APPROVED** [review]

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py [hunk 1]
```diff
@@ -170,6 +171,39 @@ def check_valid_value(self, value, config_type=None):
...
+                checked. Defaults to None. None means all PVs will be compared.
+            neg_pattern (_type_, optional): PVs which match this pattern will
+                not be checked. Defaults to None. None means no PV will be
+                excluded from comparison. In case of conflict with
+                `pos_pattern`, it will take precedence.
+        """
+        pos_re = _re.compile('' if pos_pattern is None else pos_pattern)
+        # In case of None in neg_pattern, create a regexp that never matches:
+        # https://stackoverflow.com/questions/1723182/a-regex-that-will-never-be-matched-by-anything
+        neg_re = _re.compile(r'(?!x)x' if neg_pattern is None else pos_pattern)
```
Comment (anacso17) [inline siriuspy/siriuspy/clientconfigdb/configdb_client.py:193]: bug? `neg_re = _re.compile(r'(?!x)x' if neg_pattern is None else **neg**_pattern)`

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py [hunk 2]
```diff
@@ -170,6 +171,39 @@ def check_valid_value(self, value, config_type=None):
         config_type = self._process_config_type(config_type)
         return _templates.check_value(config_type, value)
 
+    @staticmethod
```
Comment (anacso17) [inline siriuspy/siriuspy/clientconfigdb/configdb_client.py:174]: I think this can be migrated to PVsConfig class

Comment (fernandohds564) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py:174]: I don't know. I don't usually use PVsConfig, I prefer to use this basic class.
maybe we could add a method in PVsConfig to compare the "self" with another config.
what do you think?

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py (hunk 1 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py]: done!

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py [hunk 3]
```diff
@@ -170,6 +171,39 @@ def check_valid_value(self, value, config_type=None):
...
+        """
+        pos_re = _re.compile('' if pos_pattern is None else pos_pattern)
+        # In case of None in neg_pattern, create a regexp that never matches:
+        # https://stackoverflow.com/questions/1723182/a-regex-that-will-never-be-matched-by-anything
+        neg_re = _re.compile(r'(?!x)x' if neg_pattern is None else pos_pattern)
+
+        cf1pvs = {n: v for n, v, t in config1['pvs']}
+        cf2pvs = {n: v for n, v, t in config2['pvs']}
+
+        allpvs = set(cf1pvs.keys() | cf2pvs.keys())
```
Comment (ericonr) [inline siriuspy/siriuspy/clientconfigdb/configdb_client.py:198]: For what it's worth, it seems `dict1.keys() | dict2.keys()` already returns a set, so the cast to `set`  shouldn't be necessary.

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py [hunk 4]
```diff
@@ -170,6 +171,39 @@ def check_valid_value(self, value, config_type=None):
...
+                excluded from comparison. In case of conflict with
+                `pos_pattern`, it will take precedence.
+        """
+        pos_re = _re.compile('' if pos_pattern is None else pos_pattern)
+        # In case of None in neg_pattern, create a regexp that never matches:
+        # https://stackoverflow.com/questions/1723182/a-regex-that-will-never-be-matched-by-anything
+        neg_re = _re.compile(r'(?!x)x' if neg_pattern is None else neg_pattern)
+
+        cf1pvs = {n: v for n, v, t in config1['pvs']}
+        cf2pvs = {n: v for n, v, t in config2['pvs']}
```
Comment (murilobalves) [inline siriuspy/siriuspy/clientconfigdb/configdb_client.py:196]: couldn't be `{n: v for n, v, _ in config1['pvs']}` since `t` is not being used?

**Review by murilobalves — APPROVED** [review]

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py (hunk 3 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py:198]: right, thanks!

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py (hunk 2 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py:174]: yes, in the future we could do that here!

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py (hunk 3 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py]: done!

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py (hunk 4 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py]: right, done!

File: siriuspy/siriuspy/clientconfigdb/configdb_client.py (hunk 2 above)
Comment (anacso17) [reply siriuspy/siriuspy/clientconfigdb/configdb_client.py:174]: ok!

**Review by anacso17 — APPROVED** [review]

**Review by xresende — APPROVED** [review]

**Pull request merged**